arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -0,0 +1,644 @@
+      LOG_ERROR(main_logger, "Configuration not valid.");
+      config = nullptr;
+      return FAIL;
+    }
+
+    return SUCCESS;
+  }
+
+  void initialize()
+  {
+    // Create algorithm from configuration
+    initialize_cameras_landmarks::set_nested_algo_configuration( "initializer",
+                                                                 config,
+                                                                 algorithm );
+  }
+
+  void clear_ptrs()
+  {
+    camera_map_ptr = nullptr;
+    landmark_map_ptr = nullptr;
+    feature_track_set_ptr = nullptr;
+    sfm_constraint_ptr = nullptr;
+  }
+
+  void load_tracks( )
+  {
+    kwiver::vital::path_t in_tracks_path =
+      config->get_value<kwiver::vital::path_t>("input_tracks_file");
+    feature_track_set_ptr =
+      kwiver::vital::read_feature_track_file(in_tracks_path);
+  }
+
+  void load_sfm_constraint( )
+  {
+    if(config == nullptr)
+    {
+      return;
+    }
+
+    sfm_constraint_ptr = std::make_shared<sfm_constraints>();
+    kwiver::vital::image_container_sptr first_frame;
+    if(config->has_value("video_source") &&
+       config->get_value<std::string>("video_source") != "")
+    {
+      video_input_sptr video_reader;
+      video_name = config->get_value<std::string>("video_source");
+      video_input::set_nested_algo_configuration("video_reader", config,
+                                                 video_reader);
+      video_input::get_nested_algo_configuration("video_reader", config,
+                                                 video_reader);
+      video_reader->open( config->get_value<std::string>("video_source") );
+      if (video_reader->get_implementation_capabilities()
+        .has_capability(video_input::HAS_METADATA))
+      {
+        sfm_constraint_ptr->set_metadata(video_reader->metadata_map());
+        kwiver::vital::timestamp ts;
+        video_reader->next_frame( ts );
+        first_frame = video_reader->frame_image();
+      }
+      else
+      {
+        return;
+      }
+    }
+    else
+    {
+      return;
+    }
+
+    using kwiver::vital::simple_camera_intrinsics;
+    using kwiver::vital::simple_camera_perspective;
+    using kwiver::vital::frame_id_t;
+    using kwiver::vital::metadata_sptr;
+    using kwiver::vital::intrinsics_from_metadata;
+    using kwiver::vital::local_geo_cs;
+
+    #define GET_K_CONFIG(type, name) \
+    config->get_value<type>(bc + #name, K_def.name())
+
+    simple_camera_intrinsics K_def;
+    const std::string bc = "video_reader:base_camera:";
+    auto K = std::make_shared<simple_camera_intrinsics>(
+      GET_K_CONFIG(double, focal_length),
+      GET_K_CONFIG(kwiver::vital::vector_2d, principal_point),
+      GET_K_CONFIG(double, aspect_ratio),
+      GET_K_CONFIG(double, skew));
+
+    auto baseCamera = simple_camera_perspective();
+    baseCamera.set_intrinsics(K);
+    auto md = sfm_constraint_ptr->get_metadata()->metadata();
+    kwiver::vital::camera_map::map_camera_t camMap;
```

Comment:
might be good to rename variables avoiding camelCase now that we are firmly in KWIVER land and not in a Qt app where the styles are all intermixed.